geom/src/duration.rs

```diff
@@ -196,6 +202,50 @@ impl Duration {
         (max, labels)
     }
 
+    /// Shows only the largest unit (hours, minute, seconds), rounded to `precision` decimal points.
+    ///
+    /// ```
+    /// use geom::Duration;
+    /// assert_eq!(Duration::seconds(3600.0).to_rounded_string(0), "1hr");
+    /// assert_eq!(Duration::seconds(3600.0).to_rounded_string(1), "1.0hr");
+    /// assert_eq!(Duration::seconds(7800.0).to_rounded_string(0), "2hr");
+    /// assert_eq!(Duration::seconds(800.0).to_rounded_string(1), "13.3min");
+    /// assert_eq!(Duration::seconds(-800.0).to_rounded_string(1), "-13.3min");
+    /// assert_eq!(Duration::seconds(0.0).to_rounded_string(0), "0");
+    /// assert_eq!(Duration::seconds(12.5).to_rounded_string(1), "12.5s");
+    /// assert_eq!(Duration::seconds(12.5).to_rounded_string(2), "12.50s");
+    /// ```
+    pub fn to_rounded_string(self, precision: usize) -> String {
```

Comment:
I think that this kind of rounding works well for many of our tight spaces like axis labels.

Reply:
This is pretty nice! And I forgot all about doctests, perfect way to describe the behavior.